tests/auto/tst_declarativetabmodel/tst_declarativetabmodel.cpp

```diff
@@ -1,493 +0,0 @@
-    QTest::newRow("mailto query does not count") << "mailto:joe@example.com?cc=bob@example.com&body=hello1" << "mailto2";
-}
-
-void tst_declarativetabmodel::invalidTabs()
-{
-    QSignalSpy countChangeSpy(tabModel, SIGNAL(countChanged()));
-    QSignalSpy activeTabChangedSpy(tabModel, SIGNAL(activeTabChanged(int,bool)));
-
-    QFETCH(QString, url);
-    QFETCH(QString, title);
-    int originalCount = tabModel->count();
-    tabModel->addTab(url, title, originalCount);
-
-    QCOMPARE(tabModel->count(), originalCount);
-    QCOMPARE(countChangeSpy.count(), 0);
-    QCOMPARE(activeTabChangedSpy.count(), 0);
-}
-
-void tst_declarativetabmodel::reloadModel()
-{
-    tabModel->deleteLater();
-    QTest::waitForEvents();
-
-    setTestData(EMPTY_QML);
-    setTestData(QML_SNIPPET);
-
-    tabModel = TestObject::qmlObject<DeclarativeTabModel>("tabModel");
-    QSignalSpy loadedSpy(tabModel, SIGNAL(loadedChanged()));
-    QVERIFY(tabModel);
-    waitSignals(loadedSpy, 1);
-
-    QCOMPARE(tabModel->count(), 2);
-
-    int activeTabIndex = tabModel->findTabIndex(tabModel->activeTab().tabId());
-    QModelIndex modelIndex = tabModel->createIndex(activeTabIndex, 0);
-    QCOMPARE(tabModel->activeTab().tabId(), tabModel->data(modelIndex, DeclarativeTabModel::TabIdRole).toInt());
-    QCOMPARE(tabModel->activeTab().url(), tabModel->data(modelIndex, DeclarativeTabModel::UrlRole).toString());
-    QCOMPARE(tabModel->activeTab().title(), tabModel->data(modelIndex, DeclarativeTabModel::TitleRole).toString());
-    QCOMPARE(tabModel->activeTab().currentLink(), tabModel->m_tabs.at(activeTabIndex).currentLink());
-
-    modelIndex = tabModel->createIndex(0, 0);
-    QCOMPARE(tabModel->data(modelIndex, DeclarativeTabModel::UrlRole).toString(), QString("https://sailfishos.org/sailfish-silica/index.html"));
-    QCOMPARE(tabModel->data(modelIndex, DeclarativeTabModel::TitleRole).toString(), QString("Creating applications with Sailfish Silica | Sailfish Silica 1.0"));
-    QCOMPARE(tabModel->data(modelIndex, DeclarativeTabModel::TabIdRole).toInt(), 3);
-    QCOMPARE(tabModel->m_tabs.at(0).currentLink(), 3);
-
-    modelIndex = tabModel->createIndex(1, 0);
-    QCOMPARE(tabModel->data(modelIndex, DeclarativeTabModel::UrlRole).toString(), QString("foo/bar/index.html"));
-    QCOMPARE(tabModel->data(modelIndex, DeclarativeTabModel::TabIdRole).toInt(), 4);
-    QCOMPARE(tabModel->m_tabs.at(1).currentLink(), 12);
-}
-
-void tst_declarativetabmodel::changeTabAndLoad()
-{
-    // Highest linkId of available tabs is 13
-    int nextLinkId = DBManager::instance()->nextLinkId();
-    QCOMPARE(nextLinkId, 13);
-
-    tabModel->activateTab(1);
-    QCOMPARE(currentTabId(), 4);
-
-    QCOMPARE(tabModel->activeTab().currentLink(), 12);
-    QString url = "http://www.foobar.com/something";
-    tabModel->updateUrl(currentTabId(), url, false);
-    QTest::qWait(1000);
-
-    QCOMPARE(tabModel->activeTab().tabId(), 4);
```

Comment:
```tst_persistenttabmodel::activateTabById()``` could be extended to check also url and title of the active tab when activation is expected to succeed and fail.